libraries/chain/include/scorum/chain/dba/db_accessor_i.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+#include <scorum/chain/dba/db_accessor.hpp>
+#include <scorum/chain/dba/db_accessor_mock.hpp>
+
+namespace scorum {
+namespace chain {
+namespace dba {
+
+template <typename TObject> struct db_accessor_i
+{
+    template <typename T>
+    db_accessor_i(T&& inst)
+        : _inst(std::forward<T>(inst))
+    {
+    }
+
+    using object_type = TObject;
+    using modifier_type = typename std::function<void(object_type&)>;
+    using object_cref_type = std::reference_wrapper<const TObject>;
+
+    const object_type& create(const modifier_type& modifier)
+    {
+        return _inst.visit([&](auto& x) -> decltype(auto) { return x.create(modifier); });
+    }
+
+    void update(const modifier_type& modifier)
+    {
+        _inst.visit([&](auto& x) { x.update(modifier); });
+    }
+
+    void update(const object_type& o, const modifier_type& modifier)
+    {
+        _inst.visit([&](auto& x) { x.update(o, modifier); });
+    }
+
+    void remove()
+    {
+        _inst.visit([&](auto& x) { x.remove(get()); });
+    }
+
+    void remove(const object_type& o)
+    {
+        _inst.visit([&](auto& x) { x.remove(o); });
+    }
+
+    bool is_exists() const
+    {
+        return _inst.visit([&](auto& x) -> decltype(auto) { return x.is_exists(); });
+    }
+
+    const object_type& get() const
+    {
+        return _inst.visit([&](auto& x) -> decltype(auto) { return x.get(); });
+    }
+
+    template <class IndexBy, class Key> const object_type& get_by(const Key& arg) const
+    {
+        return _inst.visit([&](auto& x) -> decltype(auto) { return x.get_by<IndexBy>(arg); });
+    }
+
+    template <class IndexBy, class Key> const object_type* find_by(const Key& arg) const
```

Comment:
I don't sure that we need in raw pointer in interface. There are 'get_by', 'is_exists_by' (and 'obtain' if required) instead.